lib/cpp/src/thrift/transport/TSSLSocket.cpp

```diff
@@ -318,6 +318,8 @@ bool TSSLSocket::peek() {
     } else if (rc == 0) {
       ERR_clear_error();
       break;
+    } else {
+      break;
```

Comment:
interesting...

Reply:
This case is omitted causing the method to always return false .